src/AdvancedHMC.jl

```diff
@@ -139,9 +143,21 @@ include("sampler.jl")
 export sample
 
 include("abstractmcmc.jl")
-export DifferentiableDensityModel
 
-include("contrib/ad.jl")
+Hamiltonian(metric::AbstractMetric, ℓ::LogDensityModel) = Hamiltonian(
+    metric,
+    Base.Fix1(LogDensityProblems.logdensity, ℓ.logdensity),
+    Base.Fix1(LogDensityProblems.logdensity_and_gradient, ℓ.logdensity),
+)
```

Comment:
```suggestion
function Hamiltonian(metric::AbstractMetric, ℓ::LogDensityModel)
    ℓπ = ℓ.logdensity
    cap = LogDensityProblems.capabilities(ℓπ)
    if cap === nothing
        throw(ArgumentError("The log density function does not support the LogDensityProblems.jl interface"))
    end
    if cap === LogDensityProblems.LogDensityOrder{0}()
        throw(ArgumentError("The gradient of the log density function is not defined: Implement `LogDensityProblems.logdensity_and_gradient` or use automatic differentiation by calling `Hamiltionian(metric, logdensity, AD; kwargs...)` where AD is one of the backends supported by LogDensityProblemsAD.jl"))
    return Hamiltonian(
        metric,
        Base.Fix1(LogDensityProblems.logdensity, ℓ.logdensity),
        Base.Fix1(LogDensityProblems.logdensity_and_gradient, ℓ.logdensity),
    )
end
```

The first check should maybe moved to AbstractMCMC.